### ISSUE TITLE
Difficulties with higher video speeds and multiple videos 

### ISSUE THREAD

**Issue by WilhelmFleig** [olivierfriard/BORIS#649]
Dear Mr. Friard,

I am currently experiencing issues with higher video playback speeds. Overall, I have noticed that the new version (version 8) has some problems with handling multiple videos, resulting in increased processor usage. Currently, I am working with 8 videos at 1080p resolution simultaneously, and this is causing lags in display. The CPU regularly reaches 100% utilization, which also causes the program to become unresponsive or very slow. For instance, pausing a video sometimes takes up to 5 seconds. It becomes even more challenging when using higher video playback speeds. At 1.5x speed, the program usually doesn't respond to any actions I take. These issues were not present in version 7, where I also had decent GPU utilization, often running at 100% at higher speeds, but it never became unresponsive.

I am not entirely sure what is causing these issues, but I have some ideas that might be contributing. I am not sure how it's implemented, but there may be some "internal" problem, possibly related to the periodic spikes in CPU usage that are new to version 8. On the other hand, it seems to me that there is no regulation of the frame rate. When doubling the speed, the GPU should be less affected if it only needs to display every second frame. Is there any automatic frame rate regulation implemented?

I hope this provides you with a good insight into my current problem. If not, please let me know.

The system that I am currently using is as follows:
Windows 10
Version 22H2
Razer Blade Stealth 13 (Version 2020)
16 Gb Ram 
Intel Core i765G7
eGPU RTX 3070 
currently using Boris Version 8.20.7 
same problem also comes with version 8.20.4 
comparing to version 7.13.9

Best regards,
Wilhelm Fleig

**Comment by olivierfriard** [comment]
Hi,
thank you for reporting this.
Did you try to change the "MPV player hardware video decoding" parameter?
See Project > Preferences

**Comment by WilhelmFleig** [comment]
Hi, 
thanks for the quick response.
Yes i've tried all of the MPV settings, but the issues still accure.

**Comment by olivierfriard** [comment]
Hi,
can you send me info about your video (See Tools > Media information) in order to reproduce the problem?

**Comment by WilhelmFleig** [comment]
Hi, 
here you can see the media information:



MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.0
Duration: 3864.052244897959
FPS: 24.999662399291992
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3864.05224 seconds (01:04:24.052)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 96601306
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.2 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3864.05 (01:04:24.050)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 37.804433333333336
Duration: 3860.634875283447
FPS: 29.96916389465332
Video size: 1920x1080
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3860.634867 seconds (01:04:20.635)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1920x1080
Number of frames: 115819046
Bitrate: 6.0 Mb 
FPS: 29.97002997002997
Has video: True
Has audio: True
File size: 2.9 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3860.64 (01:04:20.640)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.16
Duration: 3777.6522448979595
FPS: 24.99965476989746
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3777.65224 seconds (01:02:57.652)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 94441306
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.1 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3777.65 (01:02:57.650)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.04
Duration: 3776.2122448979594
FPS: 24.99965476989746
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3776.21224 seconds (01:02:56.212)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 94405306
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.1 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3776.21 (01:02:56.210)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.0
Duration: 3772.8464400000003
FPS: 24.999692916870117
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3772.84644 seconds (01:02:52.846)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 94321161
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.1 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3772.85 (01:02:52.850)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.0
Duration: 3772.8464400000003
FPS: 24.999692916870117
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3772.84644 seconds (01:02:52.846)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 94321161
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.1 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3772.85 (01:02:52.850)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.0
Duration: 3767.092244897959
FPS: 24.999652862548828
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3767.09224 seconds (01:02:47.092)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 94177306
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.1 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3767.09 (01:02:47.090)


MPV information
Video format: h264
Number of media in media list: 1
Current time position: 0.0
Duration: 3766.612244897959
FPS: 24.999652862548828
Video size: 1280x720
Aspect ratio: 1.778

ffprobe analysis
File path: blacked out

Duration: 3766.61224 seconds (01:02:46.612)
Format long name: QuickTime / MOV
Creation time: NA
Resolution: 1280x720
Number of frames: 94165306
Bitrate: 2.3 Mb 
FPS: 25.0
Has video: True
Has audio: True
File size: 1.1 Gb
Video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Audio codec: AAC (Advanced Audio Coding)

Total duration: 3766.61 (01:02:46.610)

**Comment by WilhelmFleig** [comment]
Dear Mr. Friard, 
I have tested BORIS a bit further and came to notice that all my problems are solved when running BORIS only on the internal GPU (Intel(R) Iris(R) Xe Graphics). This caused a higher CPU use, but the GPU was no longer running at 100% all the time. Instead, it was only running at around 36%. So maybe there are some issues with Nvidia GPUs. Is this adjustable?

**Comment by olivierfriard** [comment]
Dear Wilhelm,
thank you for reporting this. How did you disable the Nvidia GPU?
Can you please try the experimental version v.8.22.1 (just released on https://github.com/olivierfriard/BORIS/releases/tag/v8.22.1) with and without the Nvidia GPU?
Thank you for your collaboration

**Comment by WilhelmFleig** [comment]
Dear Mr. Friard,
To change the GPU, I opened the NVIDIA Control Panel and then navigated to the 3D Settings where you can select the "Manage 3D settings" tab. There, I chose internal GPU as the preferred global setting.
I have tested the new BORIS version you provided, and the issue appears to be fixed. When using the Nvidia GPU, both the CPU and GPUs were utilized at a moderate speed (for 2.5X video speed: CPU at 70%, GPU at 60%). The videos showed no lag, and it seems to be working fine. When using only the internal GPU, the CPU usage increased slightly, but the GPU surprisingly had a low usage of about 40% (for 2X video speed: CPU at 85%, GPU at 50%).
In conclusion, the problem appears to be resolved, and the Nvidia GPU seems to be detected properly, with no program shutdowns anymore.
Thank you very much for your support.
